flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/source/KafkaSourceITCase.java

```diff
@@ -299,10 +319,17 @@ public void deserialize(
             if (deserializer == null) {
                 deserializer = new IntegerDeserializer();
             }
-            collector.collect(
-                    new PartitionAndValue(
-                            new TopicPartition(record.topic(), record.partition()),
-                            deserializer.deserialize(record.topic(), record.value())));
+
+            if (enableObjectReuse) {
+                reuse.tp = new TopicPartition(record.topic(), record.partition()).toString();
+                reuse.value = deserializer.deserialize(record.topic(), record.value());
+                collector.collect(reuse);
+            } else {
+                collector.collect(
+                        new PartitionAndValue(
+                                new TopicPartition(record.topic(), record.partition()),
+                                deserializer.deserialize(record.topic(), record.value())));
+            }
         }
```

Comment:
I don't quite understand the purpose of this. Could you explain it a little bit?  thank you.

Reply:
Thanks for the review! Basically the bug is caused by using object-reusing deserialization schema in KafkaSource. Some explanation can be found in [FLINK-25132](https://issues.apache.org/jira/browse/FLINK-25132).

I improved this test case for validating that KafkaSource indeed works well if deserialization schema reuses object.